src/PluginForm/Bluesnap/HostedPaymentFieldsPaymentMethodAddForm.php

```diff
@@ -135,6 +148,10 @@ protected function buildCreditCardForm(
       '#weight' => -200,
     ];
 
+    // Add bluesnap device datacollector iframe.
+    $mode = $this->entity->getPaymentGateway()->getPluginConfiguration()['mode'];
```

Comment:
**CHANGE REQUEST**

We have an API function specifically for getting the BlueSnap configuration, use that.
`$mode = $this->entity->getPaymentGateway()->getBluesnapConfig()['mode'];`

Reply:
Used the bluesnap config